nylas/models/connectors.py

```diff
@@ -0,0 +1,162 @@
+from dataclasses import dataclass
+from typing import Dict, Any, List, Optional, Union
+from typing_extensions import TypedDict, NotRequired
+
+from dataclasses_json import dataclass_json
+
+from nylas.models.auth import Provider
+from nylas.models.list_query_params import ListQueryParams
+
+
+@dataclass_json
+@dataclass
+class Connector:
+    """
+    Interface representing the Nylas connector response.
+
+    Attributes:
+        name: Custom name of the connector
+        provider: The provider type
+        settings: Optional settings from provider
+        scope: Default scopes for the connector
+    """
+
+    name: str
+    provider: Provider
+    settings: Optional[Dict[str, Any]] = None
+    scope: Optional[List[str]] = None
+
+
+class BaseCreateConnectorRequest(TypedDict):
+    """
+    Interface representing the base Nylas connector creation request.
+
+    Attributes:
+        name: Custom name of the connector
+        provider: The provider type
+    """
+
+    name: str
+    provider: Provider
+
+
+class GoogleCreateConnectorSettings(TypedDict):
+    """
+    Interface representing a Google connector creation request.
+
+    Attributes:
+        client_id: The Google Client ID
+        client_secret: The Google Client Secret
+        topic_name: The Google Pub/Sub topic name
+    """
+
+    client_id: str
+    client_secret: str
+    topic_name: NotRequired[str]
+
+
+class MicrosoftCreateConnectorSettings(TypedDict):
+    """
+    Interface representing a Microsoft connector creation request.
+
+    Attributes:
+        client_id: The Google Client ID
+        client_secret: The Google Client Secret
+        tenant: The Microsoft tenant ID
+    """
+
+    client_id: str
+    client_secret: str
+    tenant: NotRequired[str]
+
+
+class GoogleCreateConnectorRequest(BaseCreateConnectorRequest):
+    """
+    Interface representing the base Nylas connector creation request.
+
+    Attributes:
+        name (str): Custom name of the connector
+        provider (Provider): The provider type, should be Google
+        settings: The Google OAuth provider credentials and settings
+        scope: The Google OAuth scopes
+    """
+
+    settings: NotRequired[GoogleCreateConnectorSettings]
```

Comment:
Hey @mrashed-dev do you know why this is not required? I would assum the GoogleCreateConnectorSettings to be something that is required to create a Connector

Reply:
@kraju3 ah you're right, in the docs for some reason settings is not required but client_id and client_secret is. Looks like a typo on the docs side. Made the change now.